### PR TITLE
Upgrade to UCX 1.14.0

### PR DESCRIPTION
**Pull request by abellina**
Closes: https://github.com/NVIDIA/spark-rapids/issues/7691

This is in draft mode since UCX 1.14.0 is not out yet, but it's close. I'll need to update a few things before we can merge, since I have used a temporary version of jucx and I am pulling UCX rc6 in our dockerfile, for testing purposes.

This updates: the jucx jar version, our jenkins dockerfile, documentation, NOTICE-binary, and sample dockerfiles.

### CODE REVIEW

**Comment by pxLi** [comment]
build

**Comment by abellina** [comment]
build

**Comment by sameerz** [comment]
UCX 1.14.0 is released at https://github.com/openucx/ucx/releases/tag/v1.14.0

**Comment by abellina** [comment]
We are missing a non rc6 jar in maven central, and I've checked with UCX team and they will publish a "1.14" jar there to be our official jar soon. For now, I've updated the jenkins dockerfile to pull the right ucx tarball for ubuntu.

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
build

File: docs/additional-functionality/shuffle-docker-examples/Dockerfile.ubuntu_no_rdma [hunk 1]
```diff
@@ -16,24 +16,31 @@
...
-RUN apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
+RUN CUDA_UBUNTU_VER=`echo "$UBUNTU_VER"| sed -s 's/\.//'` && \
+  apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu$CUDA_UBUNTU_VER/x86_64/3bf863cc.pub
 
 RUN apt update
 RUN apt-get install -y wget
-RUN cd /tmp && wget https://github.com/openucx/ucx/releases/download/v$UCX_VER/ucx-v$UCX_VER-ubuntu18.04-mofed5-cuda$UCX_CUDA_VER.deb
+RUN cd /tmp && \
+  wget https://github.com/openucx/ucx/releases/download/v$UCX_VER/ucx-$UCX_VER-ubuntu$UBUNTU_VER-mofed5-cuda$UCX_CUDA_VER.tar.bz2 && \
+  tar -xvf ucx-$UCX_VER-ubuntu$UBUNTU_VER-mofed5-cuda$UCX_CUDA_VER.tar.bz2
```
[the remark this answers is not in the document]
Comment (abellina) [reply docs/additional-functionality/shuffle-docker-examples/Dockerfile.ubuntu_no_rdma]: Taken care of here: https://github.com/NVIDIA/spark-rapids/pull/7877/commits/60992a229174eab11fccd74f4766d8826414d61c

File: docs/additional-functionality/shuffle-docker-examples/Dockerfile.ubuntu_rdma [hunk 2]
```diff
@@ -19,39 +19,55 @@
...
+ARG UBUNTU_VER
 
 COPY --from=rdma_core /*.deb /tmp/
 
 RUN apt update
 RUN apt-get install -y wget
-RUN cd /tmp && wget https://github.com/openucx/ucx/releases/download/v$UCX_VER/ucx-v$UCX_VER-ubuntu18.04-mofed5-cuda$UCX_CUDA_VER.deb
+RUN cd /tmp && \
+  wget https://github.com/openucx/ucx/releases/download/v$UCX_VER/ucx-$UCX_VER-ubuntu$UBUNTU_VER-mofed5-cuda$UCX_CUDA_VER.tar.bz2 && \
+  tar -xvf ucx-$UCX_VER-ubuntu$UBUNTU_VER-mofed5-cuda$UCX_CUDA_VER.tar.bz2
```
[the remark this answers is not in the document]
Comment (abellina) [reply docs/additional-functionality/shuffle-docker-examples/Dockerfile.ubuntu_rdma]: Also taken care of in the last commit. Thank you!

**Review by jlowe — APPROVED** [review]

**Comment by abellina** [comment]
build

**Comment by abellina** [comment]
@GaryShen2008 mind giving this a look? the PR requested a number of code owner reviewers automatically, but I did want to make sure you were aware of the change.

**Comment by abellina** [comment]
actually I am not entirely sure how many code owners I'll need..

**Comment by abellina** [comment]
Ok I see it's just not a green button ;) I will merge now so we start getting runs in CI with this.

**Pull request merged**